doc/source/bugs.adoc

```diff
@@ -0,0 +1,193 @@
+
+== Core dumps
+
+If the server, or one of the accompanying programs core dumps, then you
+should rebuild the server as follows:
+
+....
+$ ./configure --enable-developer
+$ make
+$ make install
+....
+
+and then run the program again. You may have to to enable core dumps,
+via:
+
+....
+$ ulimit -c unlimited
+....
+
+When it core dumps, do:
+
+....
+$ gdb /path/to/executable /path/to/core/file
+....
+
+Enable logging in `+gdb+` via the following commands:
+
+....
+(gdb) set logging file gdb-radiusd.log
+(gdb) set logging on
+....
+
+and follow the instructions in the proceeding section.
+
+You can also enable the ``panic_action'' given in
+`+raddb/radiusd.conf+`. See the comments in that file for more details
+about automatically collecting gdb debugging information when the server
+crashes.
+
+== Debugging a live server
+
+If you can’t get a core dump, or the problem doesn’t result in a core
+dump, you may have to run the server under gdb. To do this, ensure that
+you have symbols in the binaries (i.e. a non-stripped binary) by
+re-building the server as described in the previous section. Then, run
+the server under gdb as follows:
+
+....
+$ gdb radiusd
+....
+
+Enable logging in `+gdb+` via the following commands:
+
+....
+(gdb) set logging file gdb-radiusd.log
+(gdb) set logging on
+....
+
+Tell `+gdb+` to pass any necessary command-line arguments to the server:
+
+....
+(gdb) set args ...
+....
+
+Where the ``…'' are the command-line arguments you normally pass to
+radiusd. For debugging, you probably want to do:
+
+....
+(gdb) set args -fxx
+....
+
+Then, do:
+
+....
+(gdb) run
+....
+
+When something interesting happens, you can hit CTRL-C in the window,
+and you should be back at the gdb prompt:
+
+....
+(gdb)
+....
+
+And follow the instructions in the next section.
+
+== Obtaining useful information
+
+Once you have a core dump loaded into gdb, or FreeRADIUS running under
+gdb, you may use the commands below to get useful information about the
+state of the server.
+
+If the server was built with threads, you can do:
+
+....
+(gdb) info threads
+....
+
+Which will give you information about the threads. If the server isn’t
+threaded, that command-line will print a message saying so.
+
+Then, do:
+
+....
+(gdb) thread apply all bt full
+....
+
+If the server isn’t threaded, the ``thread apply'' section isn’t
+necessary
+
+The output should be printed to the screen, and also sent to the
+gdb-radiusd.log file.
+
+You should then submit the information from the log file, along with any
+server output, the output of `+radiusd -xv+`, and information about your
+operating system to:
+
+https://bugs.freeradius.org/
+
+Submitting it to the bug database ensures that the bug report won’t get
+forgotten, and that it will be dealt with in due course.
+
+You should provide the issue number in any mail sent to the user’s list.
+
+== Valgrind
+
+On Linux systems, ``valgrind'' is a useful tool that can catch certain
```

Comment:
we shouldn't use "smart quotes":   ``valgrind''

Just `"valgrind"` is good enough.

Reply:
done